common/interface.go

```diff
@@ -18,6 +18,12 @@ type NumNodesDTO struct {
 	MaxLevel   int
 }
 
+// TrieNodesChannels defines the channels that are being used when integrating with trie nodes
+type TrieNodesChannels struct {
```

Comment:
Maybe rename to TrieIteratorChannels / TrieTraverserChannels ?

Reply:
done